app/src/main/java/com/mapbox/dash/example/MainActivity.kt

```diff
@@ -444,6 +463,18 @@ class MainActivity : DrawerActivity() {
             controller.stopNavigation()
         }
 
+        bindSwitch(
+            switch = menuBinding.customCompassDataInputs,
+            state = setCustomCompassDataInputs,
+        ) { enabled ->
+            sampleSensorEventManager?.compassData?.observeWhenStarted(this) {
+                if (enabled) {
+                    Log.d(TAG, "Updating compass data: $it")
+                    Dash.controller.updateCompassData(it)
+                }
+            }
+        }
```

Comment:
This launches a new coroutine every time a switch is checked or unchecked. Should be fixed, e.g. like this:
```suggestion
        menuBinding.customCompassDataInputs.setOnCheckedChangeListener { _, isChecked ->
            setCustomCompassDataInputs.value = isChecked
        }

        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                setCustomCompassDataInputs.collectLatest { enabled ->
                    if (!enabled) return@collectLatest
                    sampleSensorEventManager?.compassData?.collect {
                        Log.d(TAG, "Updating compass data: $it")
                        Dash.controller.updateCompassData(it)
                    }
                }
            }
        }
```